pandas/io/excel.py

```diff
@@ -465,7 +465,7 @@ def _parse_cell(cell_contents, cell_typ):
                 if not squeeze or isinstance(output[asheetname], DataFrame):
                     output[asheetname].columns = output[
                         asheetname].columns.set_names(header_names)
-            except StopIteration:
+            except EmptyDataError:
```

Comment:
well now you are ALWAYS raising ``EmptyDataError``, but before it would raise ``StopIteration`` AND ``ValueError``. so you are now gloming those together, that is NOT the purpose. you need to have 2 except blocks. This is similar to the idea of checking the messages on an exception but much cleaner.